### ISSUE TITLE
Inconsistent Type Hints: ParentClass async method vs Subclass sync implementation

### ISSUE THREAD

**Issue by MakonnenMak** [python/mypy#16970]
**Bug Report**

In some ParentClass, we have a  method `foo` is defined as asynchronous using async def. However, in some subclass, the implementation of the foo method is not marked as asynchronous. Despite this inconsistency, mypy does not raise a Type Error.

(A clear and concise description of what the bug is.)

**To Reproduce**

```python
class ParentClass:
    async def foo(self):
        raise NotImplementedError("Subclasses must implement the 'foo' method.")

class BadSubClass(ParentClass):
    def foo(self):
        return {'hi':22}

class GoodSubclass(ParentClass):
    async def foo(self):
        return {'bye': 33} 
        
async def main():
    bad_sub_instance = BadSubClass()
    result = await bad_sub_instance.foo()
    
    good_sub_instance = GoodSubclass()
    result = await good_sub_instance.foo()
    print(f"Result from GoodSubClass: {result}")
```

**Expected Behavior**

MyPy reports a type error for incorrect implementation of `foo` in `BadSubClass`.

Possibility is MyPy is smart enough to know the function isn't truly async ? But calling await on it still breaks so would expect for there to be an error.  

**Actual Behavior**

No errors reported 

**Your Environment**

Using latest MyPy

**Comment by MakonnenMak** [comment]
Think I was too quick to report, this exists with
`Check that awaitable return value is used [unused-awaitable]`

Closing this issue.
